### ISSUE TITLE
`scan_pyarrow_dataset(..).fetch(streaming=True)`  OOM

### ISSUE THREAD

**Issue by jmakov** [pola-rs/polars#11152]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars
from pyarrow import dataset


DATETIME_FROM = "2023-04-04 00:00:00"
DATETIME_TO ="2023-04-04 00:01:00"
expression_filter_datetime = polars.col("timestamp").is_between(datetime.datetime.fromisoformat(DATETIME_FROM), datetime.datetime.fromisoformat(DATETIME_TO))

ds = dataset.dataset("/data/", format="ipc")

polars.scan_pyarrow_dataset(ds) \
    .filter(expression_filter_datetime) \
    .select(["col_1"]) \
    .fetch(streaming=True)
```

### Issue description

Querying a local Arrow dataset causes memory spikes when calling `.fetch(streaming=True`. With selecting 1 column, the spikes are negligible, but when reading e.g. 30 columns, `polars` consumes +40GB RAM (from 40GB available on the node). After the spike mem consumption goes of course down. But even with 40GB RAM, the node doesn't survive the spike.

### Expected behavior

From the docs my understanding is we can process data larger than available memory using `.collect or .fectch(streaming=True)`

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.2
Index type:          UInt32
Platform:            Linux-6.1.51-1-MANJARO-x86_64-with-glibc2.38
Python:              3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         2.2.1
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              2023.9.0
matplotlib:          3.7.2
numpy:               1.24.4
pandas:              2.1.0
pyarrow:             13.0.0
pydantic:            1.10.12
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

**Comment by surister** [comment]
Streaming is only supported in certain situations, see https://pola-rs.github.io/polars-book/user-guide/concepts/streaming/#when-is-streaming-available

**Comment by jmakov** [comment]
Yes. My understanding is that `.filter()` is supported, that's why the ticket. Or am I missing something?

**Comment by ion-elgreco** [comment]
> Yes,  `.filter()`  and `.select()` are on the list hence the issue. Or am I missing something?

Is this local or are you accessing a cloud storage?

**Comment by jmakov** [comment]
Local files.

**Comment by jmakov** [comment]
So this one is a real blocker. Any help would be very welcome.

**Comment by deanm0000** [comment]
One note, fetch is for debugging, you probably want to use collect

That being said, polars's ability to process bigger than memory data doesn't mean you can collect/fetch it all into memory. Instead of collecting you can try sink_parquet` to save/stream the results to a file. In any event there's no guarantee that any bigger than memory operation (even those that are supported) will just work. It's more like if it's bigger than memory then sometimes (maybe most of the time) it can stream those results in a way that doesn't require full memory.

**Comment by jmakov** [comment]
"One note, fetch is for debugging, you probably want to use collect"
Thanks. But it's the same with `.collect(streaming=True)` - +40GB of memory and then had to restart to kernel. 

For another task I'm writing to Arrow and a few Weeks ago when I checked, `sink_ipc` wasn't working at all (can't remember the exception).

**Comment by coastalwhite** [comment]
Fetch no-longer exists.
